main.cpp

```diff
@@ -104,144 +100,91 @@ struct DoubleType
         value = nullptr;
     }
 
-    DoubleType& add(double d) 
-    { 
-        *value += d; 
-        return *this; 
-    }
-    DoubleType& subtract(double d) 
-    { 
-        *value -= d; 
-        return *this; 
-    }
-    DoubleType& multiply(double d) 
-    { 
-        *value *= d; 
-        return *this; 
-    }
-    DoubleType& divide(double d) 
-    { 
-        if(d == 0) { std::cout << "Warning: dividing by 0 results in undefined value!" << std::endl; }
-        *value /= d; 
-        return *this; 
-    }
-
-    // add/subtract/multiply/divide IntType 
+    DoubleType& add(double d);
+    DoubleType& subtract(double d);
+    DoubleType& multiply(double d);
+    DoubleType& divide(double d);
 
-    DoubleType& add(const IntType& it);
-    DoubleType& subtract(const IntType& it);
-    DoubleType& multiply(const IntType& it);
-    DoubleType& divide(const IntType& it);
+    operator double() { if(value) return *value; }
```

Comment:
you know `value` is not a `nullptr` because of how the class is constructed. 
- [ ] remove the check to clear the warning.